app/connectors/CustomsFinancialsApiConnector.scala

```diff
@@ -160,6 +168,71 @@ class CustomsFinancialsApiConnector @Inject()(httpClient: HttpClientV2,
       logger.error(s"Unable to download CSV :${e.getMessage}")
       Left(UnknownException)
   }
+
+  def postCashAccountStatementRequest(eori: String,
+                                      can: String,
+                                      from: LocalDate,
+                                      to: LocalDate)
+                                     (implicit hc: HeaderCarrier): Future[Either[ErrorResponse, AccountResponseCommon]] = {
+
+    val request = CashAccountStatementRequestDetail(eori, can, from.toString, to.toString)
+
+    httpClient.post(url"$retrieveCashAccountStatementsUrl")
+      .withBody[CashAccountStatementRequestDetail](request)
+      .execute[AccountResponseCommon]
+      .map(processStatusCode)
+
+  }.recover {
+    case UpstreamErrorResponse(_, BAD_REQUEST, _, _) =>
+      logger.error("BAD Request for postCashAccountStatements")
+      Left(BadRequest)
+
+    case UpstreamErrorResponse(_, INTERNAL_SERVER_ERROR, _, _) =>
+      logger.error("No Transactions available for postCashAccountStatements")
+      Left(NoTransactionsAvailable)
```

Comment:
Create a new case object like InternalServerErrorException (adding exception will differentiate from the play InterServerError status code) and use here as there is no NoTransactionsAvailable) for this call. Same for Service Unavailable. Also use the correct log messages. 

Once done.update the tests accordingly